### ISSUE TITLE
Average::mode throws for non-integer values

### ISSUE THREAD

**Issue by rgvy** [markrogoyski/math-php#266]
Average::mode throws for non-integer values

More specifically, the function uses array_count_values which throws an E_WARNING (see http://php.net/manual/en/function.array-count-values.php). 

Maybe my php isn't configured properly, but this causes a full crash. May I suggest this function to return NaN if non-integer values are detected?

Steps to reproduce:
```
$mode = Average::mode([1, 1.5, 2, 2]);
```

**Comment by markrogoyski** [comment]
Hi,

Thanks for your interest in MathPHP and reporting this issue.

I was able to reproduce the issue with the input you provided. Thanks for providing the input.

I've modified the ```Average::mode``` to work with non-integer number values. I've added your test case and a few others to the unit tests to validate that it works as expected.

The change is currently in the _develop_ branch. This change will be included in the next release, v0.41.0.

Thanks again for reporting this issue.
Mark
